### PR TITLE
fix: modify some tests

### PR DESCRIPTION
**Pull request by wep21**
A part of #434
ros2 porting of #348
* CI: Targeting Noetic.

* CI: Trying to fix build crash.

* Fixing new linter errors.

* Changing command for running tests.

Signed-off-by: wep21 <border_goldenmarket@yahoo.co.jp>

### CODE REVIEW

**Comment by wep21** [comment]
FYI @clalancette @JWhitleyWork

**Comment by wep21** [comment]
@clalancette @JWhitleyWork friendly ping

File: velodyne_pointcloud/tests/CMakeLists.txt [hunk 1]
```diff
@@ -10,10 +10,18 @@ ament_target_dependencies(test_row_step
   velodyne_msgs
 )
 
-# C++ gtests
-# catkin_add_gtest(test_calibration test_calibration.cpp)
-# add_dependencies(test_calibration ${catkin_EXPORTED_TARGETS})
-# target_link_libraries(test_calibration velodyne_rawdata ${catkin_LIBRARIES})
+ament_add_gtest(test_calibration
+  test_calibration.cpp
+  ../src/lib/calibration.cpp
```
Comment (JWhitleyWork) [inline velodyne_pointcloud/tests/CMakeLists.txt:15]: Relative file locations should not be used. If you require a file outside of this folder, please move these tests into the root CMakeLists.txt file. This is the common setup anyway.

File: velodyne_pointcloud/tests/test_calibration.cpp [hunk 2]
```diff
@@ -147,43 +146,9 @@ TEST(Calibration, hdl64e_s21)
...
-  EXPECT_EQ(laser.min_intensity, 0);
-}
-
 // Run all the tests that were declared with TEST()
 int main(int argc, char ** argv)
 {
   testing::InitGoogleTest(&argc, argv);
-  init_global_data();
   return RUN_ALL_TESTS();
 }
```
Comment (JWhitleyWork) [inline velodyne_pointcloud/tests/test_calibration.cpp:154]: The whole main function is no longer necessary. `ament_add_gtest` now links against the `GTest_Main` library which eliminates the need for the `main` function.

**Review by JWhitleyWork — CHANGES_REQUESTED** [review]
See comments.

File: velodyne_pointcloud/tests/CMakeLists.txt (hunk 1 above)
Comment (wep21) [reply velodyne_pointcloud/tests/CMakeLists.txt]: addressed at https://github.com/ros-drivers/velodyne/pull/452/commits/b420863c74a320e1c19e7e611c013151d08f022d

File: velodyne_pointcloud/tests/test_calibration.cpp (hunk 2 above)
Comment (wep21) [reply velodyne_pointcloud/tests/test_calibration.cpp]: addressed at https://github.com/ros-drivers/velodyne/pull/452/commits/bbd8a9afd2fbc911ce7a2540946e72c010000d25

**Comment by wep21** [comment]
@JWhitleyWork friendly ping

**Comment by wep21** [comment]
@JWhitleyWork friendly ping

File: velodyne_pointcloud/tests/test_calibration.cpp [hunk 3]
```diff
@@ -146,44 +145,3 @@ TEST(Calibration, hdl64e_s21)
...
-  EXPECT_EQ(laser.min_intensity, 0);
-
-  // check similar values for the last laser:
-  laser = calibration.laser_corrections[63];
-  EXPECT_FALSE(laser.two_pt_correction_available);
-  EXPECT_FLOAT_EQ(laser.vert_correction, -0.20683046990039078);
-  EXPECT_FLOAT_EQ(laser.horiz_offset_correction, -0.025999999);
-  EXPECT_EQ(laser.max_intensity, 255);
-  EXPECT_EQ(laser.min_intensity, 0);
-}
```
Comment (JWhitleyWork) [inline velodyne_pointcloud/tests/test_calibration.cpp:181]: Why remove this test?

**Review by JWhitleyWork — COMMENTED** [review]
One last question

File: velodyne_pointcloud/tests/test_calibration.cpp (hunk 3 above)
Comment (wep21) [reply velodyne_pointcloud/tests/test_calibration.cpp:181]: I restore the test in https://github.com/ros-drivers/velodyne/pull/452/commits/96707883e0632b1fd278efe712b070ed6e3767c2.

**Review by JWhitleyWork — APPROVED** [review]

**Review by JWhitleyWork — APPROVED** [review]

**Pull request merged**